catalyst_voices/lib/app/view/app_precache_image_assets.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:catalyst_voices/widgets/indicators/voices_circular_progress_indicator.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+/// A widget that pre-caches SVG and image assets before displaying its
+/// child widget.
+///
+/// This widget is useful for improving the perceived performance of your
+/// app by pre-loading any necessary image assets before they are displayed on
+/// the screen.
+/// This can help to avoid stuttering or delays when the user navigates to a
+/// new screen that requires those images.
+///
+/// [AppPrecacheImageAssets] depends on [Theme] and is trying to make
+/// as little work as possible when [Theme] is changing by caching
+/// previously loaded assets.
+class AppPrecacheImageAssets extends StatefulWidget {
+  /// The child widget to be displayed once the images have been pre-cached.
+  final Widget child;
+
+  const AppPrecacheImageAssets({
+    super.key,
+    required this.child,
+  });
+
+  @override
+  State<AppPrecacheImageAssets> createState() => _AppPrecacheImageAssetsState();
+}
+
+class _AppPrecacheImageAssetsState extends State<AppPrecacheImageAssets> {
+  final _svgs = <SvgGenImage>[];
+  final _assets = <AssetGenImage>[];
+
+  bool _hadImages = false;
+
+  late Future<void> _cacheFuture;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final theme = Theme.of(context);
+
+    final svgs = [
```

Comment:
Would it make sense that instead of hardcoding the assets to cache - we we able to pass them as an argument to the widget? Otherwise it's not reusable

Reply:
Well, yes, i may do it